mnefun/_mnefun.py

```diff
@@ -1792,19 +1792,26 @@ def _viz_raw_ssp_events(p, subj, ridx):
                 if ri in p.get_projs_from]
     all_proj = op.join(pca_dir, 'preproc_all-proj.fif')
     projs = read_proj(all_proj)
-    ev_names = [op.join(pca_dir + '/' + ii) for ii in
-                ['preproc_ecg-eve.fif', 'preproc_blink-eve.fif']]
-    ev = [read_events(e) for e in ev_names]
-    assert len(ev) == 2
-    ev = np.concatenate((ev[0], ev[1]))
-    ev = ev[np.argsort(ev[:, 0], axis=0)]
+    ev = []
+    for n in ['ecg', 'blink']:
+        fname = op.join(pca_dir, 'preproc_%s-eve.fif' % n)
+        if op.isfile(fname):
+            ev.append(read_events(fname))
+    if len(ev) == 2:
+        ev = np.concatenate((ev[0], ev[1]))
+        ev = ev[np.argsort(ev[:, 0], axis=0)]
+        colors = {int(idd): c for c, idd in zip(['r', 'b'], np.unique(ev[:, 2]))}
+    else:
+        ev = ev[0]
+        colors = {int(np.unique(ev[:, 2])[0]): 'r'}
```

Comment:
sine you already have the loop, it's better just to do all the processing there. For example:
```
colors = dict()
ev = np.zeros((0, 3), int)
for n in [...]:
    if op.isfile(...)
        ev = np.concatenate((ev, ...))
        colors[...] = ...
ev = ev[np.argsort(ev[:, 0], axis=0)]
```
No extra conditionals would be needed in this case. You also *shouldn't* need the `if len(colors) == 0` check later, either, since `mne-python` should properly handle zero-length `events` (if it doesn't it should be fixed at the `mne-python` end)